x-pack/plugins/security_solution/server/lib/detection_engine/signals/types.ts

```diff
@@ -44,8 +44,15 @@ export interface SignalSource {
   [key: string]: SearchTypes;
   '@timestamp': string;
   signal?: {
-    parent: Ancestor;
+    // parent is deprecated: new signals should populate parents instead
+    // both are optional until all signals with parent are gone and we can safely remove it
+    parent?: Ancestor;
```

Comment:
`SignalSource` represents the signal results coming from ES - we need to support pre-7.10 signals which may still have `parent` or may have `parents`.